crypto/evp/m_sigver.c

```diff
@@ -251,6 +251,26 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
         }
     }
 
+    if (ctx->reqdigest != NULL
+            && !EVP_PKEY_is_a(locpctx->pkey, SN_hmac)
```

Comment:
Not sure I like this is_a() stuff here.. It feels like a hack.

Reply:
Uhm, that's actually the proper way to do this.